src/main/java/org/codehaus/plexus/archiver/zip/AbstractZipArchiver.java

```diff
@@ -136,15 +136,18 @@
 
     private static int getJavaVersion()
     {
-        String javaSpecVersion = System.getProperty( "java.specification.version" );
```

Comment:
Looks like this code proves to cause more trouble than it is worth. Once Java becomes the minimum required version `isJava7OrLower` will be deleted. I'm not sure it is worth to try to keep up with the Java versioning changes just for that. We can just change:
```java
private static final boolean isJava7OrLower = getJavaVersion() <= 7;
```
to 
```java
private static final boolean isJava7 = System.getProperty( "java.specification.version" ).startsWith( "1.7" );
```

As the Java 7 is the minimum required version we know that if it is not Java 7 then it is newer version. Also whenever the Java version string schema will be for Java version higher than 7 it will not begin with "1.7". So this code will work as long "java.specification.version" is not an empty string. We can add check for that and throw an exception - you know just in case.

Reply:
Given Java 7 is the exception, I would be reluctant to throw an exception if it's empty, but instead to try and detect 1.7 and if that fails fallback to the normal code path.